openapi/code/named.go

```diff
@@ -95,41 +95,80 @@ func (n *Named) Singular() *Named {
 	return n
 }
 
+func (n *Named) splitASCII() (w []string) {
+	return SplitASCII(n.Name)
+}
+
+// Return the value of cond evaluated at the nearest letter to index i in name.
+// dir determines the direction of search: if true, search forwards, if false,
+// search backwards.
+func search(name string, cond func(rune) bool, dir bool, i int) bool {
+	nameLen := len(name)
+	incr := 1
+	if !dir {
+		incr = -1
+	}
+	for j := i; j >= 0 && j < nameLen; j += incr {
+		if unicode.IsLetter(rune(name[j])) {
+			return cond(rune(name[j]))
+		}
+	}
+	return false
+}
+
+// Return the value of cond evaluated on the rune at index i in name. If that
+// rune is not a letter, search in both directions for the nearest letter and
+// return the result of cond on those letters.
+func checkCondAtNearestLetters(name string, cond func(rune) bool, i int) bool {
+	r := rune(name[i])
+
+	if unicode.IsLetter(r) {
+		return cond(r)
+	}
+	return search(name, cond, true, i) && search(name, cond, false, i)
+}
+
 // emulate positive lookaheads from JVM regex:
 // (?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|([-_\s])
 // and convert all words to lower case
-func (n *Named) splitASCII() (w []string) {
+func SplitASCII(name string) (w []string) {
```

Comment:
can you revert it to being a method on Named? it's not technically needed, as what is being split gets fed into Named anyway.
<img width="458" alt="image" src="https://github.com/databricks/databricks-sdk-go/assets/259697/487e20ae-5fe9-46d3-a4ab-7176e3c4b214">

Reply:
Good point on the namespacing!

If I revert to the old implementation, splitASCII fails on `K8S_FAILURE` and `Patch:Request`, producing `k8, s, failure` and `patch:, request` respectively